src/spawnCreeps/creepBodyParts.js

```diff
@@ -0,0 +1,37 @@
+for(const gameRooms in Game.rooms) {
+
+  const controllerLevel = Game.rooms[gameRooms].controller.level;
+  const extensions = Game.rooms[gameRooms].find(FIND_STRUCTURES, {
+    filter: (structure) => {
+      return (structure.structureType == STRUCTURE_EXTENSION)
+    }
+  });
+
+  if(controllerLevel === 2) {
+    var bodyParts = [WORK, CARRY, MOVE, MOVE];
+  }
+  else if(controllerLevel === 3) {
+    var bodyParts = [WORK, WORK, CARRY, MOVE, MOVE, MOVE];
+  }
+  else if(controllerLevel === 4) {
+    var bodyParts = [WORK, WORK, CARRY, CARRY, MOVE, MOVE, MOVE, MOVE];
+  }
+  else if(controllerLevel === 5) {
+    var bodyParts = [WORK, WORK, WORK, CARRY, CARRY, CARRY, MOVE, MOVE, MOVE];
+  }
+  else if(controllerLevel === 6) {
+    var bodyParts = [WORK, WORK, WORK, WORK, CARRY, CARRY, CARRY, CARRY, MOVE, MOVE, MOVE, MOVE];
+  }
+  else if(controllerLevel === 7) {
+    var bodyParts = [WORK, WORK, WORK, WORK, WORK, CARRY, CARRY, CARRY, CARRY, CARRY, MOVE, MOVE, MOVE, MOVE, MOVE];
+  }
+  else if(controllerLevel === 8) {
+    var bodyParts = [WORK, WORK, WORK, WORK, WORK, CARRY, CARRY, CARRY, CARRY, CARRY, MOVE, MOVE, MOVE, MOVE, MOVE];
+  }
+  else {
+    var bodyParts = [WORK, CARRY, MOVE, MOVE];
+  }
+}
+
+export default bodyParts;
```

Comment:
I'm kind of surprised this works and doesn't throw errors.  I would expect that `bodyParts` isn't a defined variable when trying to export it.  `bodyParts` is declared inside the scope of the for loop.  Take a look at scope:

http://omnigeek.robmiracle.com/2011/10/14/understanding-scope-for-beginning-programmers/

it's an applicable concept to pretty much every programming language.